File: ls/server/src/test/java/org/opencds/cqf/cql/ls/server/provider/ContentServiceProviderTest.java
package org.opencds.cqf.cql.ls.server.provider;

import static org.junit.jupiter.api.Assertions.*;

import java.io.IOException;
import java.io.UncheckedIOException;
import java.net.URI;
import java.util.Set;
import org.hl7.elm.r1.VersionedIdentifier;
import org.junit.jupiter.api.Test;
import org.opencds.cqf.cql.ls.core.ContentService;
import org.opencds.cqf.cql.ls.core.utility.Uris;

public class ContentServiceProviderTest {

    @Test
    void should_throwException_when_gettingLibrary() throws Exception {
        VersionedIdentifier versionedIdentifier = new VersionedIdentifier();
        versionedIdentifier.withVersion("1.0.0");

        ContentServiceSourceProvider contentServiceSourceProvider = new ContentServiceSourceProvider(
                Uris.parseOrNull("/provider/content/sample-library-1.0.0.json"), new ContentService() {
                    @Override
                    public Set<URI> locate(URI root, VersionedIdentifier identifier) {
                        throw new UncheckedIOException(new IOException());
                    }
                });

        assertThrows(RuntimeException.class, () -> contentServiceSourceProvider.getLibrarySource(versionedIdentifier));
    }
}

File: ls/server/src/test/kotlin/org/opencds/cqf/cql/ls/server/provider/ContentServiceProviderTest.kt
package org.opencds.cqf.cql.ls.server.provider

import org.hl7.elm.r1.VersionedIdentifier
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test
import org.opencds.cqf.cql.ls.core.ContentService
import org.opencds.cqf.cql.ls.core.utility.Uris
import java.io.IOException
import java.io.UncheckedIOException
import java.net.URI

class ContentServiceProviderTest {
    @Test
    fun should_throwException_when_gettingLibrary() {
        val versionedIdentifier = VersionedIdentifier()
        versionedIdentifier.withVersion("1.0.0")

        val contentServiceSourceProvider =
            ContentServiceSourceProvider(
                Uris.parseOrNull("/provider/content/sample-library-1.0.0.json")!!,
                object : ContentService {
                    override fun locate(
                        root: URI,
                        identifier: VersionedIdentifier,
                    ): Set<URI> {
                        throw UncheckedIOException(IOException())
                    }
                },
            )

        assertThrows(RuntimeException::class.java) { contentServiceSourceProvider.getLibrarySource(versionedIdentifier) }
    }
}
